tests/save_modified_properties_test.py

```diff
@@ -34,24 +41,35 @@ def test_save_modified_properties():
 
         # Create a workspace
         workspace = Workspace(h5file_path)
+        points = Points.create(workspace)
+        workspace.finalize()
 
-        with patch(
-            "geoh5py.shared.Entity.allow_move", new_callable=PropertyMock
-        ) as allow_move:
-            allow_move.return_value = True
-            points = Points.create(workspace, vertices=xyz)
-            points.vertices = xyz * 2
+        assert write_attributes.called, f"{write_attributes} was not called."
+        assert (
+            not write_coordinates.called
+        ), f"{write_coordinates} should not have been called."
+        assert (
+            not write_data_values.called
+        ), f"{write_data_values} should not have been called."
 
-            workspace.finalize()
+        points.vertices = xyz
+        assert (
+            "vertices" in points.modified_attributes
+        ), "'vertices' should be in list of 'modified_attributes' "
 
-            assert (
-                len(allow_move.mock_calls) == 1
-            ), "Attributes re-written along with vertices."
+        workspace.finalize()
 
-            points.allow_delete = False
+        assert write_coordinates.called, f"{write_coordinates} should have been called."
+        assert (
+            not write_data_values.called
+        ), f"{write_data_values} should not have been called."
 
-            workspace.finalize()
+        points.add_data({"rando": {"values": np.ones(n_data)}})
 
-            assert (
-                len(allow_move.mock_calls) == 2
-            ), "Not all attributes were re-written."
+        assert write_data_values.called, f"{write_data_values} should have been called."
```

Comment:
**question (non-blocking):** is there no need for a `finalize`?
unclear to me when finalize is required. Only when creating new entities?
The doc says: _Finalize the h5file by checking for **updated** entities_